src/main/java/com/monumental/controllers/MonumentController.java

```diff
@@ -211,21 +217,53 @@ public MonumentBulkValidationResult validateMonumentCSV(@ModelAttribute BulkCrea
     }
 
     /**
-     * Create monuments from csv or zip
+     * Start the job to create monuments from csv or zip
      * @param request - Contains the field mapping and the file to process
-     * @return BulkCreateResult - Object representing the results of the Bulk Monument Create operation
+     * @return AsyncJob - Object containing the Id of the job created and the current value of the Future object
      */
-    @PostMapping(value = "/api/monument/bulk/create")
-    public List<Monument> createMonumentCSV(@ModelAttribute BulkCreateMonumentRequest request) throws IOException {
+    @PostMapping("/api/monument/bulk/create/start")
+    public AsyncJob startBulkCreateMonumentJob(@ModelAttribute BulkCreateMonumentRequest request) throws IOException {
         BulkCreateMonumentRequest.ParseResult parseResult = request.parse(this.monumentService);
 
         MonumentBulkValidationResult validationResult = this.monumentService.validateMonumentCSV(
                 parseResult.csvContents, parseResult.mapping, parseResult.zipFile
         );
 
-        return this.monumentService.bulkCreateMonuments(
+        /* TODO: This is not a particularly easy way of creating AsyncJobs, I can't think of a way to abstract
+         * it away currently because the AsyncJob must be passed to the CompletableFuture method, and the CompletableFuture
+         * must be passed to the AsyncJob, making it difficult to do so dynamically
+         */
+        AsyncJob job = this.asyncMonumentService.createJob();
+        job.setFuture(this.asyncMonumentService.bulkCreateMonuments(
+                job,
                 new ArrayList<CsvMonumentConverterResult>(validationResult.getValidResults().values())
-        );
+        ));
+        return job;
+    }
+
+    /**
+     * Check the progress of a create bulk monuments job
+     * @param id - Id of the job to check
+     * @return AsyncJob - Object containing the Id of the job and the current value of the Future object
+     */
+    @GetMapping("/api/monument/bulk/create/progress/{id}")
+    public AsyncJob getBulkCreateMonumentJob(@PathVariable Integer id) {
+        return this.asyncMonumentService.getJob(id);
```

Comment:
Here, we retrieve the started job by its Id, so that the front end can check if it's done yet. This is something that will be run on an interval by the UI until it sees that the job has completed.

What gets returned here looks something like:
```
{
    "id": 1,
    "future": {
        "done": false, // Or true if it's done
        "cancelled": false,
        "completedExceptionally": false,
        "numberOfDependents": 0
    },
    "progress": 0.65 // Or some other percent, or 1.0 if it's done (although the "done" field should be used to check this)
}
```